rpe/resources/gcp.py

```diff
@@ -1114,3 +1087,73 @@ def _get_request_args(self):
                 self._resource_data["name"],
             ),
         }
+
+
+class GcpDataformRepository(GoogleAPIResource):
+    service_name = "dataform"
+    resource_path = "projects.locations.repositories"
+    version = "v1beta1"
+
+    required_resource_data = ["name", "location", "project_id"]
+
+    resource_components = {
+        "iam": "getIamPolicy",
+    }
+
+    resource_type = "dataform.googleapis.com/Repository"
+
+    inferred_data_map = {
+        "uniquifier": "createTime",
+    }
+
+    def _get_request_args(self):
+        return {
+            "name": "projects/{}/locations/{}/repositories/{}".format(
```

Comment:
can the string format be extracted out to its own method and reused, same for workspace as well?